### PR TITLE
DEV: Change `tag` type to `tags` type for theme object schema

### PR DESCRIPTION
**Pull request by tgxworld**
## Why this change?

While working on the tag selector for the theme object editor, I
realised that there is an extremely high possibility that users might want to select
more than one tag. By supporting the ability to select more than one
tag, it also means that we get support for a single tag for free as
well.

## What does this change do?

1. Change `type: tag` to `type: tags` and support `min` and `max`
   validations for `type: tags`.

2. Fix the `<SchemaThemeSetting::Types::Tags>` component to support the
   `min` and `max` validations
 
 ## Recordings 
 
Kapture 2024-03-22 at 13 44 10

### CODE REVIEW

File: app/assets/javascripts/admin/addon/components/schema-theme-setting/types/tag.gjs [hunk 1]
```diff
@@ -1,26 +0,0 @@
-import Component from "@glimmer/component";
```
Comment (tgxworld) [inline app/assets/javascripts/admin/addon/components/schema-theme-setting/types/tag.gjs:1]: Renamed to `tags.gjs`

**Comment by nattsw** [comment]
LINTTTT *shakes fist at IDE*

**Comment by tgxworld** [comment]
@nattsw lol this is some locale lint that I have never seen before.

File: app/assets/javascripts/admin/addon/components/schema-theme-setting/types/tags.gjs [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import { and, not } from "truth-helpers";
+import I18n from "discourse-i18n";
+import FieldInputDescription from "admin/components/schema-theme-setting/field-input-description";
+import TagChooser from "select-kit/components/tag-chooser";
+
+export default class SchemaThemeSettingTypeTag extends Component {
```
Comment (nattsw) [inline app/assets/javascripts/admin/addon/components/schema-theme-setting/types/tags.gjs:9]: Hmm I'm not sure if you want to pluralise this. 

Like, tags chooser vs tag chooser.

Comment (tgxworld) [reply app/assets/javascripts/admin/addon/components/schema-theme-setting/types/tags.gjs]: O yea I should now that it has been renamed.

**Review by nattsw — APPROVED** [review]

**Comment by tgxworld** [comment]
Thank you @nattsw 👍

**Pull request merged**